### ISSUE TITLE
Recommended documentation addition

### ISSUE THREAD

**Issue by mvolkmann** [w0rp/ale#249]
To remove the gutter background color, add this to .vimrc:
````
highlight clear ALEErrorSign
highlight clear ALEWarningSign
````
This is especially useful when using emojis for error and warning signs that are configured like this:
````
let g:ale_sign_error = '💣'
let g:ale_sign_warning = '⚠'
````

**Comment by liuchengxu** [comment]
Hi, I also have written a colorscheme to avoid this situation :).  

[space-vim-dark](https://github.com/liuchengxu/space-vim-dark)

<img width="48" alt="2017-01-27 7 25 13" src="https://cloud.githubusercontent.com/assets/8850248/22402853/50083194-e640-11e6-94be-f92fc1b5564b.png">

**Comment by w0rp** [comment]
I have added this to the FAQ now: https://github.com/w0rp/ale#faq-change-signs

**Comment by joaqo** [comment]
@liuchengxu do you mind sharing how you got those red and yellow dots using your colorscheme? I like the colorscheme I am currently using, but would love to add those dots to it.

**Comment by liuchengxu** [comment]
@joaqo  Very simple. As mentioned in https://github.com/w0rp/ale#faq-change-signs , what you need to do is to change the default signs in your .vimrc:

```vim
let g:ale_sign_error = '•'
let g:ale_sign_warning = '•'
```

Concerning the colors, you can customize them to your personal taste.

```vim
hi link ALEErrorSign    Error
hi link ALEWarningSign  Warning
```

**Comment by joaqo** [comment]
Hey @liuchengxu thanks for the answer. When I run
```
hi link ALEErrorSign    Error
hi link ALEWarningSign  Warning
```
For some reason the background of the icon turns red and the icon turns black, which is kind of the inverse of what I was hoping so, very weird. Do you have any idea of how could I be messing this up?

**Comment by liuchengxu** [comment]
@joaqo Check the current highlight setting via `:verbose hi ALEErrorSign`. In my case, it says:

```vim
:verbose hi ALEErrorSign
ALEErrorSign   xxx links to Error
        Last set from ~/.vim/plugged/space-vim-dark/colors/space-vim-dark.vim
```

next step, `:verbose hi Error`:

```vim
:verbose hi Error
Error          xxx cterm=bold ctermfg=160 ctermbg=235 gui=bold guifg=#e0211d
        Last set from ~/.vim/plugged/space-vim-dark/colors/space-vim-dark.vim
```

Based on [256-xterm-24bit-rgb-color-chart](http://www.calmar.ws/vim/256-xterm-24bit-rgb-color-chart.html), I can see the colors are all expected. In terms of your problem, I think you should take a look at cterm option of Error. Does it have been set as `cterm=inverse`?

**Comment by joaqo** [comment]
`:verbose hi ALEErrorSign` looks the same as yours, but `:verbose hi Error` looked like this:
```
:verbose hi Error
Error          xxx term=reverse cterm=bold,reverse ctermfg=167 ctermbg=235 gui=bold,reverse guifg=#fb4934 guibg=bg
        Last set from ~/.vim/plugged/gruvbox/colors/gruvbox.vim
```

So it seems there is some wierd configs going on in my color theme gruvbox. Tweaked things a bit and ended up using,
```
hi link ALEErrorSign    GruvboxRed
hi link ALEWarningSign  GruvboxYellow
```
which might help anyone else using gruvbox and works great. @liuchengxu thanks for the help!

**Comment by liuchengxu** [comment]
@joaqo Glad you have figured it out.

The reverse comes from `cterm=bold,reverse gui=bold,reverse` in the highlight setting of Error. If someone still wants to *fix* the reverse issue, just remove `reverse`:

```
hi link ALEErrorSign    Error
hi link ALEWarningSign  Warning

hi Error   cterm=bold gui=bold
hi Warning cterm=bold gui=bold
```

For the case of gruvbox, it's simpler to use joaqo's solution: `GruvboxRed` and `GruvboxYellow`.

**Comment by her** [comment]
Just to add, if you want to avoid linking colors defined by your `colorscheme` (like Error or Warning) you can define the colors yourself like so:

```vim
let g:ale_sign_error = "◉"
let g:ale_sign_warning = "◉"
highlight ALEErrorSign ctermfg=9 ctermbg=15 guifg=#C30500 guibg=#F5F5F5
highlight ALEWarningSign ctermfg=11 ctermbg=15 guifg=#ED6237 guibg=#F5F5F5
```
`gui` sets `gvim` and `cterm` sets `vim`
`fg`, `bg` set the 'foreground' and 'background' colors respectively

<img width="307" alt="screen shot 2018-02-25 at 8 34 57 pm" src="https://user-images.githubusercontent.com/10278096/36654015-89fc2140-1a6e-11e8-819a-9e05a4526844.png">

**Comment by w0rp** [comment]
Nice settings. I'm going to use those myself. It's a pity I can't make the signs prettier by default. You can't really make any assumptions about what characters someone's font might support, etc.

**Comment by her** [comment]
hi @w0rp! glad you like it 🙂

💟 thanks for the amazing plugin!! It's really great.

**Comment by realestninja** [comment]
> Just to add, if you want to avoid linking colors defined by your `colorscheme` (like Error or Warning) and don't want to redefine Error or Warning, you can define ALEErrorSign and ALEWarningSign colors yourself like so:
> 
> ```viml
> let g:ale_sign_error = "◉"
> let g:ale_sign_warning = "◉"
> highlight ALEErrorSign ctermfg=9 ctermbg=15 guifg=#C30500 guibg=#F5F5F5
> highlight ALEWarningSign ctermfg=11 ctermbg=15 guifg=#ED6237 guibg=#F5F5F5
> ```
> 
> `gui` sets `gvim` and `cterm` sets `vim`
> `fg`, `bg` set the 'foreground' and 'background' colors respectively
> 
> <img alt="screen shot 2018-02-25 at 8 34 57 pm" width="307" src="https://user-images.githubusercontent.com/10278096/36654015-89fc2140-1a6e-11e8-819a-9e05a4526844.png">

@her When I override ctermfg/ctermbg, it just removes the colors, but it doesn't use the colors of the provided number. What am I missing? I'm using [base16 Solarized Dark](https://github.com/chriskempson/base16-shell) terminal wide. Does that change the behaviour of how I reference colors with numbers?

**Comment by roachsinai** [comment]
Hi, there. I followed @her 's setting, but I got a half circle,

![image](https://user-images.githubusercontent.com/9500049/55212398-8e3d5900-522a-11e9-9770-5a6031d15b38.png)

---

If I using `gVim` I got a small circle,

![image](https://user-images.githubusercontent.com/9500049/55212467-ca70b980-522a-11e9-93bf-eb7b315592d6.png)

---

How could I get the effect like @her ?

**Comment by roachsinai** [comment]
Worked with konsole 19.04.

**Comment by Kingmk13** [comment]
> @her When I override ctermfg/ctermbg, it just removes the colors, but it doesn't use the colors of the provided number. I end up with a default fg/bg of what my terminal is using. What am I missing? I'm using [base16 Solarized Dark](https://github.com/chriskempson/base16-shell) terminal wide. Does that change the behaviour of how I reference colors with numbers?

Me too. The only 2 ways to made it works were :
- to directly modify the sign.vim (in .vim/bundle/ale/autoload/ale/);
- to not use a theme.
So I disabled my colorsheme for now. 

The output of `:verbose hi ALEWarningSign` varry...
- When I modify the highlights and apply a custom colorsheme (anything you want), I got 
`ALEWarningSign xxx cleared`
- When I dont modify the highlight and apply a custom colorsheme, I go this 
`ALEWarningSign xxx links to Todo
        Modifié la dernière fois dans ~/.vim/bundle/ale/autoload/ale/sign.vim ligne 36`
- And what works (disabling any colorshemes, but custom highlights), I got This 
`ALEWarningSign xxx ctermfg=11 ctermbg=8 guifg=#ED6237 guibg=#F5F5F5
        Modifié la dernière fois dans ~/.vimrc ligne 144`

Maybe we should open another issues for it ? (sorry if it is not the right place)

**Comment by her** [comment]
@realestninja @Kingmk13

if there are already highlights for the group you need to use `!`

```
let g:ale_change_sign_column_color = 1 highlight! ALESignColumnWithErrors ctermfg=0 ctermbg=8 guifg=#A5A5A5 guibg=#F5F5F5 
highlight! ALESignColumnWithoutErrors ctermfg=0 ctermbg=0 guifg=#A5A5A5 guibg=#F5F5F5
```

hope this helps! 🙃

**Comment by Kingmk13** [comment]
Sorry for the late respond I was busy with other install/configs...
Thanks you @her, but it doesn't work. However I didn't know the trick with the `!` :)
It only works if you put the settings in a fyletype (like c.vim or python.vim in .vim/after/ftplugins/) or if you enter the command. So apparently it needs ALE to be fully loaded to take effect.
To have my ALE's highlight group automatically customized for all fyletype, I wrote in my .vimrc : 
`autocmd VimEnter * :let g:ale_change_sign_column_color = 1 
autocmd VimEnter * :highlight! ALESignColumnWithErrors ctermfg=0 ctermbg=8 guifg=#A5A5A5 guibg=#F5F5F5
autocmd VimEnter * :highlight! ALESignColumnWithoutErrors ctermfg=0 ctermbg=0 guifg=#A5A5A5 guibg=#F5F5F5
autocmd VimEnter * :highlight! ALEErrorSign ctermfg=9 ctermbg=8 guifg=#C30500 guibg=#F5F5F5
autocmd VimEnter * :highlight! ALEWarningSign ctermfg=11 ctermbg=8 guifg=#ED6237 guibg=#F5F5F5
autocmd VimEnter * :highlight! ALEInfoSign   ctermfg=14 ctermbg=8 guifg=#ED6237 guibg=#F5F5F5
autocmd VimEnter * :highlight! ALEError ctermfg=9 ctermbg=8 guifg=#C30500 guibg=#F5F5F5
autocmd VimEnter * :highlight! ALEWarning ctermfg=11 ctermbg=8 guifg=#ED6237 guibg=#F5F5F5
autocmd VimEnter * :highlight! ALEInfo   ctermfg=14 ctermbg=8 guifg=#ED6237 guibg=#F5F5F5
`

**Comment by Domiragi** [comment]
I have the same problem with not being to set color when using a colorscheme, checking `:verbose hi ALEWarningSign` gives me the output `ALEWarningSign xxx cleared`. 
I found out that this is a problem with the colorscheme config and its location in `vimrc`. I use haishanh/night-owl.vim; one of the settings is `hi clear`, and since I load the colorscheme after I set the settings for ALE, the highlights I set earlier get cleared out. Setting the colorscheme options before setting the highlight colors for ALE fix this problem for me.
Checking several other popular colorschemes like Monokai, Dracula or Gruvbox, they all first set `hi clear`, so I assume most colorschemes have this setting and I'm commenting here for anyone who has the same problem.
